lib/src/presentations/main/screen/main_view.dart

```diff
@@ -0,0 +1,100 @@
+import 'package:auto_route/auto_route.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:jobspot/src/core/config/localization/app_local.dart';
+import 'package:jobspot/src/core/constants/constants.dart';
+import 'package:jobspot/src/presentations/main/cubit/main_cubit.dart';
+import 'package:jobspot/src/presentations/main/widgets/custom_bottom_bar.dart';
+import 'package:jobspot/src/presentations/sign_in/widgets/custom_button.dart';
+
+class MainView extends StatelessWidget {
+  const MainView({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    final cubit = context.read<MainCubit>();
+    return Scaffold(
+      bottomNavigationBar: BlocBuilder<MainCubit, MainState>(
+        buildWhen: (previous, current) =>
+            previous.currentIndex != current.currentIndex,
+        builder: (context, state) {
+          return CustomBottomBar(
+            changeTab: (index) {
+              cubit.changeIndex(index);
+              cubit.toPage(index);
```

Comment:
Should be put cubit.toPage into cubit.changeIndex.
This place only call cubit.changeIndex

Reply:
I fix it